frontend/client/__tests__/routes.test.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import { mount } from 'enzyme'
+import App from '../App'
+import { MemoryRouter } from 'react-router'
+import { rootStore } from '../src/store/model'
+import NotFound from '../src/screens/404'
+import CodeValidations from '../src/screens/CodeValidations'
+import Login from '../src/screens/Login'
+import reactRouter from 'react-router-dom'
+import ChangePasswordModal from '../src/components/ChangePasswordModal'
+
+// Make sure we don't call any Firebase functions
+jest.mock('../src/store/firebase')
+
+beforeAll(() => {
+  // Mock BrowserRouter as it overrides MemoryRouter used in test
```

Comment:
Is this really "mocking" the BrowserRouter? Or is it just setting it to something else? My understanding is that mocking something requires a call to `jest.mock()`.

Reply:
We just want BrowserRouter to not do anything or else it interferes with MemoryRouter used in the tests. Jest.mock is only needed if we need to watch what happens to the component during the test. I was just using mock as a general term for replacing the functionality of something. This idea taken from  https://medium.com/@antonybudianto/react-router-testing-with-jest-and-enzyme-17294fefd303